contrib/pkg/utils/generic.go

```diff
@@ -0,0 +1,74 @@
+package utils
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"github.com/openshift/hive/pkg/resource"
+)
+
+type releasePayload struct {
+	PullSpec string `json:"pullSpec"`
+}
+
+func DetermineReleaseImageFromSource(sourceURL string) (string, error) {
+	resp, err := http.Get(sourceURL)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	payload := &releasePayload{}
+	err = json.Unmarshal(data, payload)
+	if err != nil {
+		return "", err
+	}
+	return payload.PullSpec, nil
+}
+
+func GetResourceHelper(command string) (resource.Helper, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		log.WithError(err).Error("Cannot get client config")
+		return nil, err
+	}
+	helper := resource.NewHelperFromRESTConfig(cfg, log.WithField("command", command))
+	return helper, nil
+}
+
+func DefaultNamespace() (string, error) {
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, &clientcmd.ConfigOverrides{})
+	ns, _, err := kubeconfig.Namespace()
+	return ns, err
+}
+
+func GetPullSecret(pullSecret string, pullSecretFile string) (string, error) {
+	envPullSecret := os.Getenv("PULL_SECRET")
+	if len(envPullSecret) > 0 {
+		return envPullSecret, nil
+	}
+	if len(pullSecret) > 0 {
+		return pullSecret, nil
+	}
+	if len(pullSecretFile) > 0 {
+		data, err := ioutil.ReadFile(pullSecretFile)
+		if err != nil {
+			log.Error("Cannot read pull secret file")
```

Comment:
`GetPullSecret` should take a logger as a parameter rather than using the global logger.